src/connector/src/source/google_pubsub/split.rs

```diff
@@ -14,31 +14,50 @@
 
 use anyhow::anyhow;
 use bytes::Bytes;
+use chrono::{TimeZone, Utc};
+use google_cloud_pubsub::client::Client;
+use google_cloud_pubsub::subscription::SeekTo;
 use serde::{Deserialize, Serialize};
 
+use super::PubsubProperties;
 use crate::source::{SplitId, SplitMetaData};
 
 #[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Hash)]
 pub struct PubsubSplit {
     pub(crate) index: u32,
+    pub(crate) subscription: String,
+    pub(crate) properties: PubsubProperties,
```

Comment:
suggest change to `start_timestamp` since you use this attr to seek from mq. 
& I want to be consistent with the other connector implementations, and I want to add the stop timestamp

Reply:
Done for `start_offset`, how would we use `stop_offset` within the implementation?

Reply:
before we pull the batch, we can check if the timestamp from the latest message is greater than the `stop_offset`. If so, we can break the loop and let the Future keep pending. 
You can refer to `src/connector/src/source/dummy_connector.rs` for this part.

Reply:
resolved with the latest set of commits